tracer/src/Datadog.Trace/Ci/ITRClient.cs

```diff
@@ -0,0 +1,428 @@
+            var response = await request.PostAsync(new ArraySegment<byte>(state), MimeTypes.Json).ConfigureAwait(false);
+            var responseContent = await response.ReadAsStringAsync().ConfigureAwait(false);
+            if (response.StatusCode is < 200 or >= 300)
+            {
+                if (finalTry)
+                {
+                    try
+                    {
+                        Log.Error<int, string>("Failed to submit events with status code {StatusCode} and message: {ResponseContent}", response.StatusCode, responseContent);
+                    }
+                    catch (Exception ex)
+                    {
+                        Log.Error<int>(ex, "Unable to read response for failed request with status code {StatusCode}", response.StatusCode);
+                    }
+                }
+
+                throw new WebException($"Status: {response.StatusCode}, Content: {responseContent}");
+            }
+
+            var deserializedResult = JsonConvert.DeserializeObject<DataArrayEnvelope<CommitResponse>>(responseContent);
+            if (deserializedResult.Data is null)
+            {
+                return null;
+            }
+
+            var stringArray = new string[deserializedResult.Data.Length];
+            for (var i = 0; i < deserializedResult.Data.Length; i++)
+            {
+                var value = deserializedResult.Data[i].Id;
+                if (ShaRegex.Matches(value).Count != 1)
+                {
+                    ThrowHelper.ThrowException($"The value '{value}' is not a valid Sha.");
+                }
+
+                stringArray[i] = deserializedResult.Data[i].Id;
+            }
+
+            return stringArray;
+        }
+    }
+
+    public async Task<long> SendObjectsPackFileAsync(string commitSha, string[] commitsExceptions)
+    {
+        Log.Debug("ITR: Packing and sending delta of commits and tree objects...");
+
+        var packFiles = await GetObjectsPackFileFromWorkingDirectoryAsync(commitsExceptions).ConfigureAwait(false);
+        if (packFiles is null)
+        {
+            return 0;
+        }
+
+        var repository = await _getRepositoryUrlTask.ConfigureAwait(false);
+        var jsonPushedSha = JsonConvert.SerializeObject(new DataEnvelopeWithMeta<CommitRequest>(new CommitRequest(commitSha), repository));
+        var jsonPushedShaBytes = Encoding.UTF8.GetBytes(jsonPushedSha);
+
+        long totalUploadSize = 0;
+        foreach (var packFile in packFiles)
+        {
+            // Send PackFile content
+            Log.Information("ITR: Sending {packFile}", packFile);
+            totalUploadSize += await WithRetries(InternalSendObjectsPackFileAsync, packFile, MaxRetries).ConfigureAwait(false);
+
+            // Delete temporal pack file
+            try
+            {
+                File.Delete(packFile);
+            }
+            catch
+            {
+                // .
```

Comment:
Maybe worth a debug log?

Reply:
Log added in https://github.com/DataDog/dd-trace-dotnet/pull/3045/commits/ee3048f6cb4f6c1027a372cdedc58883b5193527